apps/explorer/src/app/components/txs/details/tx-stop-order-cancel.tsx

```diff
@@ -0,0 +1,68 @@
+import { t } from '@vegaprotocol/i18n';
+import type { BlockExplorerTransactionResult } from '../../../routes/types/block-explorer-response';
+import { MarketLink } from '../../links/';
+import type { TendermintBlocksResponse } from '../../../routes/blocks/tendermint-blocks-response';
+import { TxDetailsShared } from './shared/tx-details-shared';
+import { TableCell, TableRow, TableWithTbody } from '../../table';
+import DeterministicOrderDetails from '../../order-details/deterministic-order-details';
+import { CancelSummary } from '../../order-summary/order-cancellation';
+import Hash from '../../links/hash';
+import type { components } from '../../../../types/explorer';
+
+export type StopOrderCancellationTransaction =
+  components['schemas']['v1StopOrdersCancellation'];
+
+interface TxDetailsStopOrderCancelProps {
+  txData: BlockExplorerTransactionResult | undefined;
+  pubKey: string | undefined;
+  blockData: TendermintBlocksResponse | undefined;
+}
+
+/**
+ * Someone cancelled a stop order
+ */
+export const TxDetailsStopOrderCancel = ({
+  txData,
+  pubKey,
+  blockData,
+}: TxDetailsStopOrderCancelProps) => {
+  if (!txData || !txData.command) {
+    return <>{t('Awaiting Block Explorer transaction details')}</>;
+  }
+
+  const command: StopOrderCancellationTransaction = txData.command;
```

Comment:
**chore (blocking)**: This is incorrect. It's an example of where this heavy-handed typing isn't ideal I suppose! If you take a look at 

https://github.com/vegaprotocol/frontend-monorepo/blob/39962a02465a9d80ecf1869de2a2732f32586237/apps/explorer/src/types/explorer.d.ts#L453

you'll see that the `txData.command` contains properties depending on the type of transaction. In this case it's:

```suggestion
  const command: StopOrderCancellationTransaction = txData.command.stopOrdersCancellation;
```

Part of the challenge of these transactions is often finding an example of one on a network. Often @daunatv has done one already, but if not you have to give it a go yourself so you can see what the actual shape ends up being. I never quite trust the generated types without seeing one first-hand.

Reply:
Adding [a link ](https://explorer.fairground.wtf/txs/46E716FCC2D01BA8DE1E4B3A4230085B2E4C41950851A5F4D3146F293EE06812)to one I have done on Fairground

Reply:
thanks both!